public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AnalyticsConfig.tsx

```diff
@@ -27,6 +27,8 @@ export const AnalyticsConfig: FunctionComponent<Props> = (props: Props) => {
     ? props.options.jsonData.logAnalyticsSubscriptionId
     : props.options.jsonData.subscriptionId;
 
+  const credentialsEnabled = primaryCredentials.authType === 'clientsecret';
```

Comment:
it seems like we manually check against the strings 'clientsecret' and 'msi' a lot. Seems like it might be worthwhile to turn these into typescript enums so you could do something here like `primaryCredentials.authType === supportedAuthTypes.clientSecret` or something like that?

Reply:
`AzureAuthType` is actually [strongly-typed](https://github.com/grafana/grafana/blob/4779de9467857b80658574ace2f41ff14bcd25a5/public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts#L50), compiler will even return error if you try to add a new value and won't update the code which uses switch statements etc.